__tests__/todo.spec.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { render, fireEvent, cleanup } from '@testing-library/react-native';
+import { build, fake } from '@jackfranklin/test-data-bot';
+import { firebase as mockFirebase } from '../src/firebase/config';
+
+import Todo from '../src/screens/todo';
+
+// mockFirebase.firestore().useEmulator('localhost', 8080);
+
+const userDataBuilder = build('User Data', {
+  fields: {
+    uid: fake((f) => f.internet.password()),
+    fullName: fake((f) => f.name.findName()),
+    email: fake((f) => f.internet.email()),
+  },
+});
+const user = userDataBuilder();
+
+const dataBuilder = build('Data', {
+  fields: {
+    word: fake((f) => f.lorem.word()),
+  },
+});
+const data = dataBuilder();
+
+const addItem = (targetInput, itemText) => {
+  fireEvent.changeText(targetInput, itemText);
+  fireEvent(targetInput, 'submitEditing');
+};
+
+afterEach(() => cleanup());
+
+test('user is able to add an item to the to-do list', async () => {
+  //* render the todo list
+  const { getByText, getByPlaceholderText } = render(<Todo user={user} />);
+
+  //* type in the input field & click submits
+  addItem(getByPlaceholderText('What needs to be done?'), data.word);
+
+  //* todo item should be added to the list
+  expect(getByText(data.word)).toBeDefined();
+  expect(getByText('1 items')).toBeDefined();
+});
+
+test('user should be able to toggle to-do item', async () => {
+  //* render the todo list
+  const { getByPlaceholderText, getByLabelText } = render(<Todo user={user} />);
+
+  //* type in the input field & click submits
+  addItem(getByPlaceholderText('What needs to be done?'), data.word);
+  const itemSwitch = getByLabelText(`${data.word}-switch`);
+
+  //* toggle switch
+  fireEvent(itemSwitch, 'onValueChange', true);
+
+  //* todo item should be completed
+  expect(itemSwitch).toHaveProp('value', true);
```

Comment:
Trigger the event and check value of the switch may be something you don't want to do.
_Let's have a call._

Reply:
Now assertion check the `style` of the item text when a switch its toggle => `expect(itemText).toHaveStyle([{ textDecorationLine: 'line-through' }]);`
**Checking the colors of the switch itself won't work** cause when the final element, a `RNSwitch`, outputs it generates static attributes for colors when the switch its on and for when its off (`tintColor` & `onTintColor` )
Call Date: September 9th
See commit `#adbb9cd` (last commit at the moment of writing this message)